pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/SchemasResource.java

```diff
@@ -0,0 +1,273 @@
+            .handle((version, error) -> {
+                if (isNull(error)) {
+                    response.resume(
+                        Response.ok().entity(
+                            new DeleteSchemaResponse(version)
+                        ).build()
+                    );
+                } else {
+                    response.resume(error);
+                }
+                return null;
+            });
+    }
+
+    @POST @Path("/{property}/{cluster}/{namespace}/{topic}/schema")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Post topic schema")
+    public void postSchema(
+        @PathParam("property") String property,
+        @PathParam("cluster") String cluster,
+        @PathParam("namespace") String namespace,
+        @PathParam("topic") String topic,
+        PostSchemaPayload payload,
+        @Suspended final AsyncResponse response
+    ) {
+        validateDestinationAndAdminOperation(property, cluster, namespace, topic);
+
+        pulsar().getSchemaRegistryService().putSchemaIfAbsent(
+            buildSchemaId(property, cluster, namespace, topic),
+            SchemaData.builder()
+                .data(payload.schema.getBytes())
+                .isDeleted(false)
+                .timestamp(clock.millis())
+                .type(SchemaType.valueOf(payload.type))
+                .user(clientAppId())
+                .build()
+        ).thenAccept(version ->
+            response.resume(
+                Response.accepted().entity(
+                    new PostSchemaResponse(version)
+                ).build()
+            )
+        );
+    }
+
+    private String buildSchemaId(String property, String cluster, String namespace, String topic) {
+        return property + "_" + cluster + "_" + namespace + "_" + topic;
+    }
+
+    private void validateDestinationAndAdminOperation(String property, String cluster, String namespace, String topic) {
+        TopicName topicName = TopicName.get(
+            domain(), property, cluster, namespace, decode(topic)
+        );
+
+        try {
+            validateDestinationExists(topicName);
+            validateAdminAccessOnProperty(topicName.getProperty());
+            validateTopicOwnership(topicName, false);
+        } catch (RestException e) {
+            if (e.getResponse().getStatus() == Response.Status.UNAUTHORIZED.getStatusCode()) {
+                throw new RestException(Response.Status.NOT_FOUND, "Not Found");
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    private void validateDestinationExists(TopicName dn) {
+        try {
+            Topic topic = pulsar().getBrokerService().getTopicReference(dn.toString());
+            checkNotNull(topic);
+        } catch (Exception e) {
+            throw new RestException(Response.Status.NOT_FOUND, "Topic not found");
+        }
+    }
+
+    static class GetSchemaResponse {
```

Comment:
If this classes will have to be used in the `PulsarAdmin` Java client, they should be added in `pulsar-common` module.